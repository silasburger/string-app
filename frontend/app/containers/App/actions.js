/*
 *
 * StringPage actions
 *
 */

import {
  FETCH_POSTS,
  POSTS_RECEIVED,
  FETCH_POSTS_FAILED,
  POST_CREATED,
  CREATE_POST_FAILED,
  CREATE_POST,
} from './constants';

export function fetchPosts() {
  return {
    type: FETCH_POSTS,
  };
}
export function fetchPostsFailed(error) {
  return {
    type: FETCH_POSTS_FAILED,
    payload: error,
  };
}
export function postsReceived(posts) {
  return {
    type: POSTS_RECEIVED,
    payload: { posts },
  };
}
export function createPost() {
  return {
    type: CREATE_POST,
  };
}
export function createPostFailed(error) {
  return {
    type: CREATE_POST_FAILED,
    payload: error,
  };
}
export function postCreated(post) {
  return {
    type: POST_CREATED,
    payload: post,
  };
}
