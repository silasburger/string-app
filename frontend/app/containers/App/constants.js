/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const FETCH_POSTS = 'app/App/FETCH_POSTS';
export const POSTS_RECEIVED = 'app/App/POSTS_RECEIVED';
export const FETCH_POSTS_FAILED = 'app/App/FETCH_POSTS_FAILED';
export const CREATE_POST = 'app/App/CREATE_POST';
export const POST_CREATED = 'app/App/POST_CREATED';
export const CREATE_POST_FAILED = 'app/App/CREATE_POST_FAILED';
