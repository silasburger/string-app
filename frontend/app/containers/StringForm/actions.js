/*
 *
 * StringForm actions
 *
 */

import {
  CHANGE_STRING,
  CREATE_POST,
  CREATE_POST_FAILED,
  POST_CREATED,
} from './constants';

export function changeString(value) {
  return {
    type: CHANGE_STRING,
    value,
  };
}
export function createPost(data) {
  return {
    type: CREATE_POST,
    data,
  };
}
export function createPostFailed(error) {
  return {
    type: CREATE_POST_FAILED,
    error,
  };
}
export function postCreated(post) {
  return {
    type: POST_CREATED,
    post,
  };
}
