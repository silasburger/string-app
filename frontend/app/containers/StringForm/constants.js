/*
 *
 * StringForm constants
 *
 */

export const CHANGE_STRING = 'app/StringForm/CHANGE_STRING';
export const CREATE_POST = 'app/StringForm/CREATE_POST';
export const POST_CREATED = 'app/StringForm/POST_CREATED';
export const CREATE_POST_FAILED = 'app/StringForm/CREATE_POST_FAILED';
