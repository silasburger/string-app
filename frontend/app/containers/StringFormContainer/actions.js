/*
 *
 * StringFormContainer actions
 *
 */

import {
  CHANGE_STRING,
  CREATE_POST,
  CREATE_POST_FAILED,
  POST_CREATED,
} from './constants';

export function changeString(value) {
  return {
    type: CHANGE_STRING,
    value,
  };
}
export function createPost() {
  return {
    type: CREATE_POST,
  };
}
export function createPostFailed(error) {
  return {
    type: CREATE_POST_FAILED,
    error,
  };
}
export function postCreated(post) {
  return {
    type: POST_CREATED,
    post,
  };
}
