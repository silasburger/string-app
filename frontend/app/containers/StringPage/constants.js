/*
 *
 * StringPage constants
 *
 */

export const CHANGE_DATE_TIME_FILTER = 'app/StringPage/CHANGE_DATE_TIME_FILTER';

/**
 * These constants are for date/time filter functionality
 */
export const ALL_POSTS = 'ALL_POSTS';
export const LAST_FIVE_MINUTES = 'LAST_FIVE_MINUTES';
export const LAST_TWO_WEEKS = 'LAST_TWO_WEEKS';
export const ONE_YEAR_OR_EARLIER = 'ONE_YEAR_OR_EARLIER';
export const FIVE_YEARS_OR_EARLIER = 'FIVE_YEARS_OR_EARLIER';
