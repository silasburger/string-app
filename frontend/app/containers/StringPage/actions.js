/*
 *
 * StringPage actions
 *
 */

import { FETCH_POSTS, POSTS_RECEIVED, FETCH_POSTS_FAILED } from './constants';

export function fetchPosts() {
  console.log('in fetch posts action');
  return {
    type: FETCH_POSTS,
  };
}
export function fetchPostsFailed(error) {
  return {
    type: FETCH_POSTS_FAILED,
    error,
  };
}
export function postsReceived(posts) {
  return {
    type: POSTS_RECEIVED,
    posts,
  };
}
