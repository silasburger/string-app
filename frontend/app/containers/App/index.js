/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import StringPage from 'containers/StringPage/Loadable';
import FormPage from 'components/FormPage';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import NavBar from '../../components/NavBar';

import GlobalStyle from '../../global-styles';

export default function App() {
  return (
    <div>
      <Helmet>
        <title>DMI Tech Screen</title>
        <meta name="DMI Tech Screen" content="An initial tech screen" />
        <noscript>Javascrirpt must be enabled for this site to work</noscript>
      </Helmet>
      <NavBar />
      <Switch>
        <Route exact path="/" component={FormPage} />
        <Route exact path="/strings" component={StringPage} />
        <Route component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </div>
  );
}
