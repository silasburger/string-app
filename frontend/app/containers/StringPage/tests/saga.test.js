/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { takeLatest, call, put, select } from 'redux-saga/effects';
import { FETCH_POSTS } from 'containers/App/constants';
import stringPageSaga, { fetchPosts } from '../saga';
import { fetchPostsFailed, postsReceived } from 'containers/App/actions';


// const generator = stringPageSaga();

describe('stringPageSaga Saga', () => {
  const stringPageGenerator = stringPageSaga();

  it('should start task to watch for FETCH_POSTS action', () => {
    const takeLatestDescriptor = stringPageGenerator.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(FETCH_POSTS, fetchPosts);
  });
});

describe('fetchPosts Saga', () => {
  let fetchPostsGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    fetchPostsGenerator = fetchPosts();

    const callDescriptor = fetchPostsGenerator.next().value;
  });

  it('should call the api', () => {
    expect(callDescriptor).toEqual(call(BackendAPI.getAllPosts));
  });

  it('should dispatch the postsReceived action if it requests the data successfully', () => {
    const posts = [{ string: 'onepost', created_at: '2019-02-16 19:10:25-07' }];
    const putDescriptor = fetchPostsGenerator.next(posts).value;
    expect(putDescriptor).toEqual(put(postsReceived(res.posts)));
  });

  it('should call the fetchPostsFailed action if the response errors', () => {
    const error = new Error('Some error');
    const putDescriptor = fetchPostsGenerator.throw(error).value;
    expect(putDescriptor).toEqual(put(fetchPostsFailed(error)));
  });
});
