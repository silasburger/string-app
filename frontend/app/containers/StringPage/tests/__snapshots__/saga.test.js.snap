// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPosts Saga should call the api 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`fetchPosts Saga should call the fetchPostsFailed action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`fetchPosts Saga should dispatch the postsReceived action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;
